### PR TITLE
Precision of POI for iOS

### PR DESCRIPTION
**Pull request by ivanPyrohivskyi**
Using precisionXY field from OBF.proto OsmAndPoiBoxDataAtom for set more precised coordinates for POI
Before:
Screenshot 2021-02-17 at 15 13 31

After:
Screenshot 2021-02-17 at 15 11 35

### CODE REVIEW

File: src/Utilities.cpp [hunk 1]
```diff
@@ -707,3 +707,18 @@ OsmAnd::PointD OsmAnd::Utilities::getTileEllipsoidNumberAndOffsetY(int zoom, dou
     res.y = floor(((xTilesCountForThisZoom / 2 - M2 * xTilesCountForThisZoom / 2 / M_PI) - yTileNumber) * tileSize);
     return res;
 }
+
+std::pair<int, int> OsmAnd::Utilities::calculateFinalXYFromBaseAndPrecisionXY(int bazeZoom, int finalZoom, int precisionXY,
+                                                                              int xBase, int yBase, bool ignoreNotEnoughPrecision) {
+    int finalX = xBase;
+    int finalY = yBase;
```
Comment (PaulStets) [inline src/Utilities.cpp:714]: int precisionCalc = precisionXY; and subsequent Java code is missing here.

File: src/Utilities.cpp [hunk 2]
```diff
@@ -707,3 +707,18 @@ OsmAnd::PointD OsmAnd::Utilities::getTileEllipsoidNumberAndOffsetY(int zoom, dou
...
+    for (int zoom = bazeZoom; zoom < finalZoom; zoom++) {
+        if (!ignoreNotEnoughPrecision) {
```
Comment (PaulStets) [inline src/Utilities.cpp:716]: The condition does not correspond to Java. + before failing you should probably output the reason for the failure.

File: src/Utilities.cpp [hunk 3]
```diff
@@ -707,3 +707,18 @@ OsmAnd::PointD OsmAnd::Utilities::getTileEllipsoidNumberAndOffsetY(int zoom, dou
     res.y = floor(((xTilesCountForThisZoom / 2 - M2 * xTilesCountForThisZoom / 2 / M_PI) - yTileNumber) * tileSize);
     return res;
 }
+
+std::pair<int, int> OsmAnd::Utilities::calculateFinalXYFromBaseAndPrecisionXY(int bazeZoom, int finalZoom, int precisionXY,
+                                                                              int xBase, int yBase, bool ignoreNotEnoughPrecision) {
```
Comment (PaulStets) [inline src/Utilities.cpp:712]: Please format the code according to C formatting style (curly brace is placed on a separate line) here and everywhere else throughout the commit.

File: src/Data/ObfPoiSectionReader_P.cpp [hunk 4]
```diff
@@ -870,6 +874,7 @@ void OsmAnd::ObfPoiSectionReader_P::readAmenity(
     QHash<int, QVariant> stringOrDataValues;
     auto categoriesFilterChecked = false;
     const CollatorStringMatcher matcher(query, StringMatcherMode::CHECK_STARTS_FROM_SPACE);
+    uint32_t precision = 0;
```
Comment (PaulStets) [inline src/Data/ObfPoiSectionReader_P.cpp:877]: It is better to name this variable exactly as in Java code.

File: src/Data/ObfPoiSectionReader_P.cpp [hunk 5]
```diff
@@ -22,6 +22,10 @@
 #include <Logging.h>
 
 const int BUCKET_SEARCH_BY_NAME = 5;
+const int BASE_POI_SHIFT = 7;
+const int PRECISED_POI_SHIFT = 5;
```
Comment (PaulStets) [inline src/Data/ObfPoiSectionReader_P.cpp:26]: PRECISED_POI_SHIFT rename to FINAL_POI_SHIFT, PRECISED_POI_ZOOM = FINAL_POI_ZOOM. (By the way, it's precise not precised)

**Comment by ivanPyrohivskyi** [comment]
Added fixes according requirement

**Pull request merged**